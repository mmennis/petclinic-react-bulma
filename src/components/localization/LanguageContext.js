import React from 'react'

export const locales = {
    en: { name: 'English', code: 'en', lang: 'English', flag: 'EN'},
    es: { name: 'Spanish', code: 'es', lang: 'Espanol', flag: 'ES'},
    fr: { name: 'French', code: 'fr', lang: 'Francais', flag: 'FR'}
}

export const LanguageContext = React.createContext({
    lang: 'en',
    updateLocale: (lang) => {},
})