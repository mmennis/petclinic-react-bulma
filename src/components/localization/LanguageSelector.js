import React from 'react'
import { locales, LanguageContext } from './LanguageContext'
import { Form } from 'react-bulma-components'
import ReactCountryFlag from 'react-country-flag'

export default class LanguageSelector extends React.Component {

    render() {
        return(
            <LanguageContext.Consumer>
                { value => (
                    <div>
                        
                        <Form.Field horizontal={true}>
                            <Form.Label size="small">Lang:</Form.Label>
                            <Form.Control>
                                <Form.Select size="small" color="primary" marginless={true} value={value.lang} onChange={value.updateLocale} data-testid="select">
                                    <option value="en">English</option>
                                    <option value="es">Espanol</option>
                                    <option value="fr">Francais</option>
                                </Form.Select>
                            </Form.Control>                        
                            <ReactCountryFlag countryCode={locales[value.lang].flag} svg style={{ 'height': '2em', 'width': '2em' }}/>
                        </Form.Field>
                    </div>
                    
                )}
            </LanguageContext.Consumer>
        )
    }
}

LanguageSelector.contextType = LanguageContext