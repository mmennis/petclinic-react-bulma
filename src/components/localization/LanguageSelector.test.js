import React from 'react'
import { LanguageContext } from './LanguageContext'
import LanguageSelector from './LanguageSelector'
import { create } from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react'

const ctx = {
    lang: 'en',
    updateLocale: jest.fn()
}
const renderLanguageSelector = () => {
    return render(
        <LanguageContext.Provider value={ctx} >
            <LanguageSelector />
        </LanguageContext.Provider>
    )
}

describe('LanguageSelector', () => {

    beforeEach(() => {
        jest.clearAllMocks()
    })

    it('should match the saved snapshot', () => {
        const langSelector = create(<LanguageSelector />)
        expect(langSelector).toMatchSnapshot()
    })

    it('should call the updateLocal method in the context', () => {
        const { getByTestId } = renderLanguageSelector()
        fireEvent.change(getByTestId('select'), {target: { value: 'fr'}})
        expect(ctx.updateLocale).toHaveBeenCalled()
        expect(ctx.lang).toBe('en')
    })

    it('should update the context if a valid updateLocale method is available', () => {
        ctx.updateLocale = (e) => {
            ctx.lang = e.target.value
        }
        const { getByTestId } = renderLanguageSelector()
        fireEvent.change(getByTestId('select'), {target: { value: 'fr'}})
        expect(ctx.lang).toBe('fr')
    })
    
})