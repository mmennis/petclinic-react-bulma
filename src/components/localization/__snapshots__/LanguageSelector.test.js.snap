// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector should match the saved snapshot 1`] = `
<div>
  <div
    className="field is-horizontal"
  >
    <label
      className="label is-small"
    >
      Lang:
    </label>
    <div
      className="control"
    >
      <div
        className="select is-small is-primary"
      >
        <select
          className="is-marginless"
          data-testid="select"
          disabled={false}
          multiple={false}
          onChange={[Function]}
          readOnly={false}
          value="en"
        >
          <option
            value="en"
          >
            English
          </option>
          <option
            value="es"
          >
            Espanol
          </option>
          <option
            value="fr"
          >
            Francais
          </option>
        </select>
      </div>
    </div>
    <img
      src="https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.4.3/flags/4x3/us.svg"
      style={
        Object {
          "display": "inline-block",
          "height": "2em",
          "verticalAlign": "middle",
          "width": "2em",
        }
      }
    />
  </div>
</div>
`;
